book/standard_library.md

```diff
@@ -14,32 +18,218 @@ The standard library currently includes:
 To see a complete list of the commands available in the standard library, run the following:
 
 ```nu
-use std
-scope commands
-| where name =~ '^std '
-| select name usage extra_usage
-| wrap "std-lib"
+nu -c "
+  use std
+  scope commands
+  | where name =~ '^std '
+  | select name description extra_description
+  | wrap 'Standard Library Commands'
+  | table -e
+"
+```
+
+::: note
+The `use std` command above loads the entire standard library so that you can see all of the commands at once. This is typically not how it will be used (more info below). It is also run in a separate Nu subshell simply so that it is not loaded into scope in the shell you are using.
+:::
+
+## Importing the Standard Library
+
+While working at the commandline, it can be convenient to load the entire standard library using:
+
+```nu
+use std *
+```
+
+However, this form should be avoided in custom commands and scripts since it has the longest load time. While the load time is only on the order of ~20-30 milliseconds on an average system, that can add up in certain scenarios.
+
+::: important Optimal Startup when Using the Standard Library
+See the [notes below](#optimal-startup) on how to ensure that your configuration isn't loading the entire Standard Library.
+:::
+
+### Importing Submodules
+
+Each submodule of the standard library can be loaded separately. Again, for best performance, load only the submodule(s) that you need in your code.
+
+There are several forms that can be used:
+
+#### 1. Import the submodule
+
+Examples:
+
+```nu
+use std/iter
+[2 5 "4" 7] | iter filter-map {|it| $it ** 2}
+
+use std/help
+help ls
+help commands
+```
+
+This form requires that you prefix the command using the submodule name. This can be useful in two scenarios.
+
+1. In the `use std/iter` example, it's important to prefix the commands with the `iter` namespace. If, on the other hand, you were to `use std/iter *`, you would shadow the built-in `find` command with the `iter` module's `find` command.
+
+2. In the `use std/help` example, `help` is both the submodule name and the name of the main command it exports. Using `use std/help` allows you to access:
+
+   ```
```

Comment:
i can never remember if these code fences should be `nu` or `nushell`
```suggestion
   ```nu
```

Reply:
Either works, but the usage through the docs is inconsistent.  I started using `nushell` for my first few PRs, but found that most existing blocks were `nu`, so I switched over to that.

```nu
ag '```nu$' | lines | length
# => 2798
ag '```nushell$' | lines | length
# => 270
```

Reply:
Pushed the fix.